### DISCUSSION TITLE
Restricting the detectable languages

### DISCUSSION THREAD

**Discussion by rohitashPrajapati** [deepgram/community#591] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

**Context**: We are facing an issue in audio transcription that, even if audio was in English but Deepgram API was resulting the transcription in Armenian language. We are currently using “detect_language=true” flag in API which detects the audio language and returns the transcription in that language.

**API:** https://api.deepgram.com/v1/listen

**Model used:**  'whisper-large'

**The solution found in the Deepgram documentation:** Deepgram specified in their documentation that we can restrict the API to return the transcription within suggested languages. For this Deepgram document says: To restrict the possible set of detectable languages, use a multivalued query parameter with the language codes as the values. For example, detect_language=en&detect_language=es will choose either English or Spanish as the detected language. Refer: https://developers.deepgram.com/docs/language-detection#restricting-the-detectable-languages

**Solutions We tried:**

1. Tried “detect_language=en&detect_language=nl” in the API, but after applying this, API don’t provide the result. [No Transcription]
2. Event tried single “detect_language=en”, API don’t respond. API only accepts “detect_language=true”. [No Transcription]
3. Tried “detect_language=true” and “language=en”, but in this case API still returns Armenian language. When “detect_language” and “language” flags are used together, than “language” flag is ignored and result of “detect_language” is used. refer: https://developers.deepgram.com/docs/language-detection#interaction-with-language-query-parameter . [Wrong Language Transcription]
4. Tried “language=en&language=nl”, but in this case API gives error: Invalid query string. Means multiple “language” flags are not allowed. [No Transcription]

**Result:** It seems, the Deepgram documents are not proper and not works as mentioned in it. Maybe the way of restricting the detectable languages only works on few limited models like “nova”, but not on Whisper models. When we tried the “detect_language=en&detect_language=nl” with "nova" model, it worked, but same with "whisper-large" is not working. So we don't have a way to restrict the transcription within suggested languages, only we can restrict to a particular language using "language" flag. 

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen?model=whisper-large&punctuate=true&diarize=true&smart_format=true&utterances=true&utt_split=0.5&detect_language=en&detect_language=nl

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

00878797-4d64-4683-926a-af153fba041e

### If possible, please attach your code or paste it into the text box.

curl --request POST \
     --url 'https://api.deepgram.com/v1/listen?model=whisper-large&punctuate=true&diarize=true&smart_format=true&utterances=true&utt_split=0.5&detect_language=en&detect_language=nl' \
     --header 'Authorization: Token xyz' \
     --header 'accept: application/json' \
     --header 'content-type: application/json' \
     --data '
{
  "url": "any english audio url"
}
'

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment, marked as answer]
Hi @rohitashPrajapati, you're correct that our language detection is only applicable to Deepgram's own models (Nova-2, Nova, Enhanced, Base). For our managed Whisper offering, Whisper has its own language detection algorithm, which can sometimes select an incorrect language as you experienced. I'm glad to hear it worked as expected with Nova. Do you otherwise prefer Whisper for its language availability, or some other factor?
